### PR TITLE
style(mypy): Enforcing typing for superset.examples module

### PR DESCRIPTION
**Pull request by john-bodley**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [x] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Adding typing enforcement for the `superset.examples` module. 

### TEST PLAN

CI. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

to: @villebro

### CODE REVIEW

File: superset/examples/countries.py [hunk 1]
```diff
@@ -2498,13 +2498,8 @@
         all_lookups[lookup][country[lookup].lower()] = country
 
 
-def get(field, symbol):
+def get(field: str, symbol: str) -> Optional[Dict[str, Any]]:
     """
     Get country data based on a standard code and a symbol
-
```
Comment (john-bodley) [inline superset/examples/countries.py:2504]: These examples were clearly wrong given the return type and thus I removed them.

File: superset/examples/helpers.py [hunk 2]
```diff
@@ -41,7 +41,7 @@
 misc_dash_slices: Set[str] = set()  # slices assembled in a 'Misc Chart' dashboard
 
 
-def update_slice_ids(layout_dict, slices):
+def update_slice_ids(layout_dict: Dict[Any, Any], slices: List[Slice]) -> None:
```
Comment (villebro) [inline superset/examples/helpers.py:44]: Not necessary a change that needs to be done here, but with proper typing, we don't necessarily need to include types in names as much in the future. Here `layout_dict` -> `layout`.

File: superset/examples/multiformat_time_series.py [hunk 3]
```diff
@@ -70,15 +73,15 @@ def load_multiformat_time_series(only_metadata=False, force=False):
...
-        "string3": ["%Y/%m/%d%H:%M:%S.%f", None],
+    dttm_and_expr_dict: Dict[str, Tuple[Optional[str], None]] = {
+        "ds": (None, None),
+        "ds2": (None, None),
+        "epoch_s": ("epoch_s", None),
+        "epoch_ms": ("epoch_ms", None),
+        "string2": ("%Y%m%d-%H%M%S", None),
+        "string1": ("%Y-%m-%d^%H:%M:%S", None),
+        "string0": ("%Y-%m-%d %H:%M:%S.%f", None),
+        "string3": ("%Y/%m/%d%H:%M:%S.%f", None),
```
Comment (villebro) [inline superset/examples/multiformat_time_series.py:84]: 👍

**Review by villebro — APPROVED** [review]

File: superset/examples/helpers.py (hunk 2 above)
Comment (john-bodley) [reply superset/examples/helpers.py:44]: @villebro I agree. I've never been a fan of variables the `_list`, `_dict`, etc. suffixes.

**Pull request merged**